packages/fast-usdc/test/fast-usdc.contract.test.ts

```diff
@@ -33,3 +45,71 @@ test('start', async t => {
   );
   t.truthy(creatorFacet);
 });
+
+test('LP deposit, withdraw', async t => {
+  const {
+    bootstrap,
+    brands: { usdc },
+    commonPrivateArgs,
+    utils,
+  } = await commonSetup(t);
+
+  const { zoe } = await setUpZoeForTest();
+  const bundle = await t.context.bundleCache.load(contractFile, contractName);
+  const installation: Installation<StartFn> = await E(zoe).install(bundle);
+
+  const { creatorFacet, publicFacet, instance } = await E(zoe).startInstance(
+    installation,
+    { USDC: usdc.issuer },
+    {
+      poolFee: usdc.make(1n),
+      contractFee: usdc.make(1n),
+    },
+    commonPrivateArgs,
+  );
+  t.truthy(creatorFacet);
+  const terms = await E(zoe).getTerms(instance);
+  const { PoolShares } = terms.brands;
+  const sharePurseP = E(terms.issuers.PoolShares).makeEmptyPurse();
+  const usdcPurseP = E(terms.issuers.USDC).makeEmptyPurse();
+
+  const { make, isGTE } = AmountMath;
+
+  {
+    const proposal = harden({
+      give: { USDC: usdc.make(100n) },
+      want: { PoolShare: make(PoolShares, 20n) },
```

Comment:
I assume a real client would need to:
- Read `shareWorth` from the contract somehow.
- Use pool-share-math.js to calculate the expected number of shares.

Does that sound realistic? If so, could we add a TODO or modify the test to follow that model more?

Reply:
Good point. Thanks for the reminder, in fact. I was thinking about this.

Reply:
That prompted me to flesh out a bunch of publish/subscribe stuff (83ed6c80a4). I like the result, though; I enhanced the story-telling aspect of the test while I was at it:

```
  ✔ LP deposits, earns fees, withdraws (1.6s)
    ℹ bootstrap vat dependencies
    ℹ Alice deposits 60  USDC
    ℹ Bob deposits 40  USDC
    ℹ Alice deposit payout 60  PoolShares
    ℹ Bob deposit payout 40  PoolShares
    ℹ contract accrues some amount of fees: 25  USDC
    ℹ Alice sees fees earned 15  USDC
    ℹ Alice withdraws 20 %: 12  PoolShares
    ℹ Bob sees fees earned 10  USDC
    ℹ Bob withdraws 80 %: 32  PoolShares
    ℹ Alice withdaw payout 15  USDC
    ℹ Bob withdaw payout 40  USDC
```

I didn't show `shareWorth` in the `t.log(...)` story, but fetching it is necessary for "Alice sees fees earned 15  USDC", which is explicit in the product spec.

To calculate the expected number of shares, `divideBy(give.USDC, shareWorth)` suffices. No need for anything special from `pool-share-math.js`.

I didn't bother to handle slippage here. I suppose `pool-share-math.js` could be enhanced to support that, though.

Reply:
Ah yea this seems more like it!

>No need for anything special from pool-share-math.js.

Ack

>I didn't bother to handle slippage here. I suppose pool-share-math.js could be enhanced to support that, though.

Hmm yea that could be tricky. The pool could also fluctuate drastically as it makes advances and receives settlements, so they'd have to call `withdraw` at the right time between advancements. Unless we wrote some logic that could put things on hold until the withdrawal completes.